api/apps/api/src/modules/access-control/scenarios-acl/scenario-access-control.ts

```diff
@@ -0,0 +1,13 @@
+import { Permit } from '../access-control.types';
+export abstract class ScenarioAccessControl {
+  abstract canEditScenario(scenarioId: string, userId: string): Promise<Permit>;
+  abstract canViewScenario(scenarioId: string, userId: string): Promise<Permit>;
+  abstract canDeleteScenario(
+    scenarioId: string,
+    userId: string,
+  ): Promise<Permit>;
+  abstract canCreateSolution(
```

Comment:
this should be the "canCreateScenario" permission. since that permission validation requires querying a different table, the service needs to depend on that other table too, but that's an implementation detail that the interface (and thus the outside world) should not care about

Reply:
what about this?